scripts/install-nix-from-closure.sh

```diff
@@ -1,181 +1,700 @@
-    cat >&2 <<EOF
+    Unable to register valid paths
+    "
+        fi
+    }
+
+    setupProfile() {
+        # Modifying current runtime environment variables to use Nix.
+        # Persistent changes are configured later.
+        print "Launching $nix/etc/profile.d/nix.sh"
+        . "$nix/etc/profile.d/nix.sh"
+        print "Left $nix/etc/profile.d/nix.sh"
+        if ! "$nix/bin/nix-env" -i "$nix"; then
+            error "
+
+    Unable to install Nix into your default profile
+    "
+        fi
+    }
+
+    installCerts() {
+        # Install an SSL certificate bundle.
+        trap 'installCertsRevert $LINENO $?' INT TERM ABRT QUIT
+        subscribeChannelsRevert() {
+            errorRevert "Received Error signal from line $1"
+            errorRevert "Reverting installation of certificate bundle"
+            $nix/bin/nix-env --uninstall "$cacert"
+            unset NIX_SSL_CERT_FILE
+            error "
+
+    Changes reverted.
+    " "$2"
+        }
+        print 'Installing SSL certificate bundle.'
+        $nix/bin/nix-env --install "$cacert"
+        export NIX_SSL_CERT_FILE="$HOME/.nix-profile/etc/ssl/certs/ca-bundle.crt"
+    }
+
+    subscribeChannels() {
+        # Subscribe the user to the Nixpkgs channel and fetch it.
+        trap 'subscribeChannelsRevert $LINENO $?' INT TERM ABRT QUIT
+        subscribeChannelsRevert() {
+            errorRevert "Received Error signal from line $1"
+            errorRevert "Reverting channels subscriptions: nixpkgs"
+            $nix/bin/nix-channel --remove nixpkgs
+            error "
+
+    Changes reverted.
+    " "$2"
+        }
+        channelUrl='https://nixos.org/channels/nixpkgs-unstable'
+        print "Subscribing to channel: nixpkgs="
+        $nix/bin/nix-channel --add "$channelUrl"
+    }
+
+    updateRepos() {
+        # Subscribe the user to the Nixpkgs channel and fetch it.
+        trap 'updateReposRevert $LINENO $?' INT TERM ABRT QUIT
+        updateReposRevert() {
+            error "
+
+    Received Error signal from line $1
+    " "$2"
+        }
+        print "Pulling information from channel: nixpkgs"
+        # nix-channel should be fully transactional, so no 'nix-channel --rollback'
+        $nix/bin/nix-channel --update nixpkgs
+    }
+
+    notice "
+
+    Performing a single-user installation of Nix:
+    "
+
+    if [ ! -d "$dest" ] ; then
+        createDest
+    fi
 
-Installation finished!  To ensure that the necessary environment
-variables are set, please add the line
+    createStore
 
-  . $p
+    copyNix
 
-to your shell profile (e.g. ~/.profile).
-EOF
-else
-    cat >&2 <<EOF
+    initializeDB
 
-Installation finished!  To ensure that the necessary environment
-variables are set, either log in again, or type
+    setupProfile
 
-  . $p
 
-in your shell.
-EOF
-fi
+    if [ -z "$NIX_SSL_CERT_FILE" ] || [ ! -f "$NIX_SSL_CERT_FILE" ]; then
+        installCerts
+    fi
+
+    if ! $nix/bin/nix-channel --list | grep -q "^nixpkgs "; then
```

Comment:
$nix needs quoting and you can replace the doubles quotes at the end by single quotes.

Reply:
Yes. Thank you.